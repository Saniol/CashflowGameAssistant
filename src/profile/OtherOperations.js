import React from 'react';
import {View, Text} from 'react-native';
import Lang from '../lang/pl';
import LayoutStyles from '../global/LayoutStyles';
import ActionRow from './ActionRow';
import ActionButton from './ActionButton';

const OtherOperations = () => (
    <View style={LayoutStyles.container}>
        <Text style={LayoutStyles.header}>{Lang.otherOperationsHeader}</Text>
        <ActionRow>
            <ActionButton color="yellow" title={Lang.otherOperationsCharity} />
            <ActionButton
                color="green"
                title={Lang.otherOperationsNewbornBaby}
            />
        </ActionRow>
        <ActionRow>
            <ActionButton color="red" title={Lang.otherOperationsFired} />
            <ActionButton
                color="purple"
                title={Lang.otherOperationsBack}
                to="/home"
            />
        </ActionRow>
        <View style={LayoutStyles.flexItem} />
        <View style={LayoutStyles.flexItem} />
    </View>
);

export default OtherOperations;
