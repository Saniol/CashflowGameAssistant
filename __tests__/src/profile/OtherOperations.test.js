import 'react-native';
import React from 'react';
import OtherOperations from '../../../src/profile/OtherOperations';
import renderWithContexts from '../../common/renderWithContexts';

describe('<OtherOperations>', () => {
    renderWithContexts(OtherOperations);
});
