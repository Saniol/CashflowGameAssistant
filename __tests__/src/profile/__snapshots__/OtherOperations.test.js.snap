// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtherOperations> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "marginTop": 12,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 12,
      }
    }
  >
    Inne
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFD024",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Dobroczynność
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C5DF41",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Narodziny dziecka
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF3822",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Zwolnienie z pracy
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8B1263",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Wróć
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
